tls/s2n_record_write.c

```diff
@@ -201,15 +215,17 @@ int s2n_record_writev(struct s2n_connection *conn, uint8_t content_type, const s
     const int is_tls13_record = cipher_suite->record_alg->flags & S2N_TLS13_RECORD_AEAD_NONCE;
     s2n_stack_blob(aad, is_tls13_record ? S2N_TLS13_AAD_LEN : S2N_TLS_MAX_AAD_LEN, S2N_TLS_MAX_AAD_LEN);
 
-    S2N_ERROR_IF(s2n_stuffer_data_available(&conn->out), S2N_ERR_BAD_MESSAGE);
+    S2N_ERROR_IF(s2n_stuffer_data_available(&conn->out), S2N_ERR_RECORD_STUFFER_NEEDS_DRAINING);
```

Comment:
`NEEDS_DRAINING` sounds like `EAGAIN` - the caller is intended to call that thing again later. But aren't outgoing buffers flushed before this is called, so this is not intended to happen. And if it happens it would be an implementation error?

Reply:
yes, likely an developer/implementation error, which is useful to have a more specific error msg for it, I'll be open to more descriptive names.